build.rs

```diff
@@ -129,7 +129,11 @@ mod llvm {
                     "create_proxy!({}; {}; {});",
                     decl.name,
                     decl.ret_ty,
-                    decl.args.trim_end_matches(',')
+                    decl.args
+                        // We cannot use `Err`, it conflicts with the Err tuple
+                        // variant.
+                        .replace("Err :", "Error :")
```

Comment:
Hm what is this about?

Reply:
If you remove that, you get an error:

```
    Updating crates.io index
   Compiling aya-rustc-llvm-proxy v0.4.0 (/home/vadorovsky/repos/rustc-llvm-proxy)
error[E0530]: function parameters cannot shadow tuple variants
   --> /home/vadorovsky/repos/rustc-llvm-proxy/target/debug/build/aya-rustc-llvm-proxy-e477d353ff005d08/out/llvm_gen.rs:920:81
    |
920 | create_proxy!(LLVMOrcLookupStateContinueLookup; (); S : LLVMOrcLookupStateRef , Err : LLVMErrorRef );
    |                                                                                 ^^^ cannot be named the same as a tuple variant
    |
   ::: /home/vadorovsky/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/prelude/mod.rs:119:13
    |
119 |     pub use super::v1::*;
    |             ------------ the tuple variant `Err` is defined here

For more information about this error, try `rustc --explain E0530`.
error: could not compile `aya-rustc-llvm-proxy` due to previous error
```

Because of this generated line in the output

```rust
create_proxy!(LLVMOrcLookupStateContinueLookup; (); S : LLVMOrcLookupStateRef , Err : LLVMErrorRef );
```

Which makes sense, that `Err :` is definitely conflicting with tuple variant `Err`. So I'm just changing it to `Error`. I think that changing the argument name shouldn't break the binding, right?

Reply:
I added more info to the comment, hopefully it's more clear now.

Reply:
I understand that we get the error, I don't understand why we get it and I don't think that the fix is in the right place?

If you look at https://github.com/aya-rs/rustc-llvm-proxy/blob/1125cfc917a650b84b759d79c5bcf70d1cb23fab/build.rs#L189 it proxies the input parameters. So presumably there's a `Err: LLVMErrorRef` in the input? And why does it work there? And if anything needs to be fixed, I think it needs to be fixed here, not later when we output the bindings.

Reply:
The real problem seems to be that using `Err` as an argument name is totally fine when declaring the function, like here:

https://gitlab.com/taricorp/llvm-sys.rs/-/blob/main/src/orc2/mod.rs#L52

But it's not fine to provide `Err` to the macro which uses that token as `$arg:ident`. To be precise, that's the signature of the macro:

https://github.com/aya-rs/rustc-llvm-proxy/blob/44d267ecb616e16dbce3b921d3fed87e065e122a/src/lib.rs#L103

I didn't fine any good solution for that yet. For sure we can't use `expr` there. I'm wondering what kind of token type would be suitable.

Reply:
If I don't figure out how to use any token type to make Rust happy with using `Err`, I think the best solution would be just replacing `Err` with something else in llvm-sys? But that will slow us down for another few days most likely.